app/views/api_views.py

```diff
@@ -20,6 +20,34 @@ def autocomplete_name(request):
         return HttpResponse('', content_type="application/json")
 
 
+@login_required(login_url='/login')
+@require_GET
+def donor_names(request):
+    try:
+        donor_names = [donor.donor_name for donor in Donor.objects.all()]
```

Comment:
see https://docs.djangoproject.com/en/4.0/ref/models/querysets/#values-list or https://docs.djangoproject.com/en/4.0/ref/models/querysets/#values for more optimized sql fetch and avoid unnecessary slower python enumeration